xdsl/transforms/stencil_to_csl_stencil.py

```diff
@@ -171,10 +295,314 @@ def match_and_rewrite(self, op: dmp.SwapOp, rewriter: PatternRewriter, /):
             nested_rewriter.rewrite_op(new_apply_op)
 
 
+def get_op_split(
+    ops: Sequence[Operation], prefetch: BlockArgument
+) -> tuple[Sequence[Operation], Sequence[Operation]]:
+    """
+    Returns a split of ops into an `(a,b)` tuple, such that:
+
+    - `a` contains neighbour accesses plus the minumum set of instructions to reduce the accessed data to 1 thing
+    - `b` contains everything else
+
+    If no valid split can be found, return `(ops, [])`.
+
+    This function does not attempt to arithmetically re-structure the computation to obtain a good split. To do this,
+    `RestructureSymmetricReductionPattern()` may be executed first.
+    """
+    a: Sequence[Operation] = []
+    b: Sequence[Operation] = []
+    rem: Sequence[Operation] = []
+    for op in ops:
+        if isinstance(op, csl_stencil.AccessOp):
+            (b, a)[op.op == prefetch].append(op)
+        else:
+            rem.append(op)
+
+    has_changes = True
+    while (
+        len(
+            a_exports := set(
+                op
+                for op in a
+                for r in op.results
+                for u in r.uses
+                if u.operation not in a
+            )
```

Comment:
I am not sure what this is computing exactly. May I suggest you put this in an aptly named helper function with a docstring? I feel this would make the code much more readable.

Reply:
This is the aptly named helper function with a docstring :)